File: pueblo-golf-tournament-api/Entities/Payment.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Entities
{
    public class Payment : BaseEntity
    {
        public string? PaymentMethod { get; set; }
        public long PaymentChannelAccountId { get; set; }
        public string? PaymentReferrencePhoto { get; set; }
        public string? AccountNumber { get; set; }
        public string? AccountName { get; set; }
        public string? ReferrenceId { get; set; }
        public DateTime PaymentDate  { get; set; }
    }
}

File: pueblo_golf_tournament_web/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';

import 'base-model.dart';
part 'payment.g.dart';

@JsonSerializable()
class Payment extends BaseModel {
  final String? paymentMethod;
  final String? referrenceId;
  final DateTime paymentDate;

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  Payment(this.paymentMethod, this.referrenceId, this.paymentDate,
      {required super.id});

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
